zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -2396,12 +2395,17 @@ private void checkDataSizeLimit(String path, byte[] data) {
   }
 
   public void watchForData(final String path) {
-    watchForData(path, false);
+    watchForData(path, false, false);
   }
 
-  private boolean watchForData(final String path, boolean skipWatchingNonExistNode) {
+  public boolean watchForData(final String path, boolean skipWatchingNonExistNode, boolean persistListener) {
     try {
-      if (skipWatchingNonExistNode) {
+      if (persistListener) {
+        retryUntilConnected(() -> {
+          ((ZkConnection) getConnection()).getZookeeper().addWatch(path, AddWatchMode.PERSISTENT);
```

Comment:
neat: can we please add more comments for this new watch type?

Reply:
sure, will add more comments